_posts/2022-mm-dd-crypto-1/2022-mm-dd-crypto1-history-and-basics.markdown

```diff
@@ -0,0 +1,157 @@
+* For as long as recorded history goes, people have felt the need to hide crucial or personal information from others. Earliest examples of what could be considered cryptography date back to Ancient Greece, India and Persia.
+* Most classical crypto algorithms relied on either swapping the order of letters or the letters themselves, up until the 15th century when so-called polyalphabetic ciphers introduced more than one possible substitution per letter. We'll talk about those in another post.
+* In the 19th century, specifically in 1883 with Auguste Kerkhoffs stating his famous principle "the enemy knows the system", the crypto world finally recognized that security _must_ come from the encryption key, not the encryption method.
+* Computers kick-started modern cryptography, because suddenly it became very easy to perform complex operations very quickly. They also moved cryptography from being something that mostly military personnel and intelligence agencies were concerned with into the academic and public domain.
+* The age of modern computing, with its ever-increasing amount of computational power (quantum computing, anyone?) poses new problems and threats to proven approaches.
+
+"But where does he want to go with this?", you may want to ask. At this point, cryptography is something that concerns us all. It's in our phones, our banks, our computers and even in our refrigerators and ceiling lamps (hopefully). And I think it's important to understand what's going on. To a degree at least. And to understand where we stand and where we're going, we need to understand where we come from. If you take anything from this paragraph, let it be "the enemy knows the system" - never trust complexity as a security device.
+
+To answer the initial question posed in the alt title: cryptography is a means of ensuring that important information can only be accessed by people/parties you're comfortable sharing it with.
+
+### Notation/Ground rules
+Before we begin, we need to agree on a few things. The first of which is "how do we actually write this down?". To keep things consistent across two authors and a number of posts, we'll define the most important details right at the start.
+
+#### Names
+Examples in the realm of cryptography always rely on names to illustrate who is interacting with whom. The most commonly used names are:
+
+**Alice**
+Alice is our "person A". She either pens the original message that starts a (hopefully encrypted) conversation or is the one that states our general problem.
+
+**Bob**
+Similarly, Bob is "person B" and usually the one to interact with Alice.
+
+**Eve**
+Eve is the **e**vil attacker that tries to interject, manipulate, listen to, ... whatever it is Alice and Bob are discussing.
+
+**Oscar**
+Oscar is our **o**pponent and, similar to Eve, tries to interfere with the communication between Alice and Bob. He might occasionally show up for the sake of flavour.
```

Comment:
These are highly opinionated but this is how I've learned it:

 - Charlie is used as a third party for protocols (but neither Alice, Bob, or Charlie are necessarily trustworthy) 
 - Eve is an eavesdropper, i.e. she's only limited to passive attacks
 - Mallory is an attacker that can abuse malleability of ciphers (i.e. schoolbook RSA or CTR-Mode)
 - Oscar is (as you said) an opponent, used as a fallback if nothing else helps or as you said for flavour.

For a trusted third party, I've rarely seen names. I've also never seen names for distinguishers (useful for mathematical assumptions like the DDH-problem).

I don't know whether it might help to think about naming a bit more for the long run.

Reply:
I'm not against adding them, I again just wanted to keep things on the simpler side to not have endless lists of things. I'll give this some more thought.